traffic_portal/app/src/common/modules/table/deliveryServices/TableDeliveryServicesController.js

```diff
@@ -17,17 +17,183 @@
  * under the License.
  */
 
-var TableDeliveryServicesController = function(deliveryServices, $scope, $state, $location, $uibModal, dateUtils, deliveryServiceUtils, locationUtils, propertiesModel) {
+var TableDeliveryServicesController = function(deliveryServices, $anchorScroll, $scope, $state, $location, $uibModal, $window, deliveryServiceService, deliveryServiceRequestService, dateUtils, deliveryServiceUtils, locationUtils, messageModel, propertiesModel, userModel) {
 
     var protocols = deliveryServiceUtils.protocols;
 
     var qstrings = deliveryServiceUtils.qstrings;
 
+    var dsRequestsEnabled = propertiesModel.properties.dsRequests.enabled;
+
     var createDeliveryService = function(typeName) {
         var path = '/delivery-services/new?type=' + typeName;
         locationUtils.navigateToPath(path);
     };
 
+    var clone = function(ds) {
+        var params = {
+            title: 'Clone Delivery Service: ' + ds.xmlId,
+            message: "Please select a content routing category for the clone"
+        };
+        var modalInstance = $uibModal.open({
+            templateUrl: 'common/modules/dialog/select/dialog.select.tpl.html',
+            controller: 'DialogSelectController',
+            size: 'md',
+            resolve: {
+                params: function () {
+                    return params;
+                },
+                collection: function() {
+                    // the following represent the 4 categories of delivery services
+                    // the ids are arbitrary but the dialog.select dropdown needs them
+                    return [
+                        { id: 1, name: 'ANY_MAP' },
+                        { id: 2, name: 'DNS' },
+                        { id: 3, name: 'HTTP' },
+                        { id: 4, name: 'STEERING' }
+                    ];
+                }
+            }
+        });
+        modalInstance.result.then(function(type) {
+            locationUtils.navigateToPath('/delivery-services/' + ds.id + '/clone?type=' + type.name);
+        }, function () {
+            // do nothing
```

Comment:
tbh, not even sure why it would fall in this function. you get a list of types, you select one and the selected one is returned. not sure why it would ever error out but rather than making this PR bigger (more risky), i'd rather do that in a follow up PR.

Reply:
maybe. Can't really say, I'm not sure what the modalInstance is. Some kind of `<dialog>` event promise?

Reply:
fixed

Reply:
right, it's a dialog that presents  you with a list of options. you select one. not sure how it would even fail ever so i removed the unused function call.